### PR TITLE
(DOCSP-29378) Upgrade Realm version for FLX permissions templates

### PR DESCRIPTION
**Pull request by nlarew**
- Upgrade to `realm@^11.8.0`
- Small updates & refactors to Realm API usage
- Update to a more robust `bluehawk.sh` file that can delete unused files & otherwise modify generated artifact projects beyond just the bluehawk command. (copied this from the web-js template apps - we should find a way to standardize this so we don't have to copypasta every time)

### CODE REVIEW

**Review by cbush — APPROVED** [review]
LGTM!

File: other/flex-sync-guides/bluehawk.sh [hunk 1]
```diff
@@ -1,9 +1,118 @@
...
+#   done
+#   jq_expr="del(${jq_expr})"
 
-rm -r generated
+#   jq "$jq_expr" package.json >> package.json.tmp
+#   mv package.json.tmp package.json
+# }
 
-for state in restricted-feed add-collaborators tiered; do
+# function remove_unused_package_json_scripts() {
```
[the remark this answers is not in the document]
Comment (nlarew) [reply other/flex-sync-guides/bluehawk.sh:44]: We use these in the web-js projects and I figure they may eventually be useful here. The `bluehawk.sh` files use the same overall structure so I figured I'd comment them out to show that the underlying structure is there.

Could delete them here but then would need to bring these back in if we need them. Thoughts?

[the remark this answers is not in the document]
Comment (cbush) [reply other/flex-sync-guides/bluehawk.sh:44]: Generally I'm a fan of just deleting stuff and re-adding as needed, but it's not a big enough deal for me to block. Do as you will!

[the remark this answers is not in the document]
Comment (nlarew) [reply other/flex-sync-guides/bluehawk.sh]: Deleted em!

**Pull request merged**